server/controllers/finance/reports/budget_analytical/index.js

```diff
@@ -201,6 +298,33 @@ async function report(req, res, next) {
 
     const totalCompletionIncome = totalRealisationIncome / totalBudgetIncome;
     const totalCompletionExpense = totalRealisationExpense / totalBudgetExpense;
+    const realisationIncomeExpense = totalRealisationIncome - totalRealisationExpense;
+
+    configurationReferences.forEach(item => {
+      item.value_account_number = 0;
+
+      item.accounts_number_formated.forEach(elt => {
+        const filterData = tabIncomeDataBudget.filter(income => {
+          return (income.number === elt);
+        });
+
+        item.value_account_number += filterData[0].realisation;
+      });
+
+    });
+
+    let firstIncomeConfigurationReferences = 0;
+    let secondIncomeConfigurationReferences = 0;
+
+    if (includeSummarySection) {
```

Comment:
Can you help me understand this block of code?  Why are the first two lines being treated differently?

And where is the `ORDER BY` or `sort()` command that guarantees the order they come out in?  Because it could be random.

Reply:
This section is really a hack, Summary section is presented as follows
![image](https://github.com/user-attachments/assets/eff81007-f888-448c-aa98-eb31c63f854c)
This section is really a hack, Summary section is presented as follows, Result equals the difference between the total of Income Total - Expenses Total, Result Without Accounts: : 71, 75, 77 and Result Without Accounts: : 71 should evaluate the result without the sum of the balance of the following accounts 71, 75 and 77 but the challenge I had was to obtain the balance of these specific accounts, that's why I had opted for using the account references by adding the Budget Analysis type account references and when creating the references I made the first grouping 71, 75 and 77 and the second 71 (also equivalent to grants)

Reply:
![image](https://github.com/user-attachments/assets/11fb686f-dc28-4086-a837-dcd8a784da9d)
To improve things, I will make the summary section dynamic, so that the analysis of the result will no longer be limited to two account references and but only in relation to the configuration of the account references

Reply:
> referen

@jniles After trying to make the summary section dynamic, I encountered the following constraint: how to distinguish the value of the subsidies represented by account 71 in the configuration?

Because in the current configuration I have configured the account references so that the subsidies are associated with the second reference, while accounts 71, 75 and 77 are grouped under the first reference. After creating a new transaction type, I make sure that this section remains compliant with the report template.

What do you think of this hack or trick, if not we will have to find another way to distinguish the subsidy accounts

![image](https://github.com/user-attachments/assets/26df2162-3727-499e-9323-e6de62a4bd7e)

Reply:
@lomamech that seems fine to me.